feat: add export curl command button for PR reviews

aaron-prindle opened the pull request:
fixes #113 

This PR adds an "Export Curl Command" button to allow for a usage pattern where users give read-only tokens but can then copy the full curl command that repo-agent would essentially run and run it locally with their own write access PAT.  That way they don't have to trust our MT setup, etc for trust.  Goal is to make security conscious users able to get most functionality possible as well as de-risk potential issue where users don't want to use repo-agent as it requires write-access PAT

Gist showing the exported command working:
https://gist.github.com/aaron-prindle/14d41d56d342bb35600acfc734c04198

<img width="2127" height="1129" alt="Screenshot 2025-11-12 4 03 16 PM" src="https://github.com/user-attachments/assets/bf244d71-472a-44ad-99b0-89c8dda9eabf" />

<img width="1989" height="692" alt="image" src="https://github.com/user-attachments/assets/4d10c8d6-60bd-4b4a-8de0-5ebd2f5c146b" />

barney-s commented:
wondering if we need an api for this ? Is there a reason this needs to be an API ? 
Can we make it part of the yaml that is returned by the PRs api ?

aaron-prindle commented:
> wondering if we need an api for this ? Is there a reason this needs to be an API ? Can we make it part of the yaml that is returned by the PRs api ?

Done, I've removed the API additions for this.

Initially I had thought of making this an API to make it analagous to "Create Draft Review" so that we could hook into any kind of "suggestion accepted" hooks we had (this would be the no write-access PAT signal for "suggestion accepted").  I agree that making it part of the returned YAML (although we need to account for diffs so it isn't purely static).  I've updated to PR here to remove the API elements.  We can revisit if we need a hook for "suggestion accepted"
